### PR TITLE
update api model and logic to use temp disk for docker root

### PR DESCRIPTION
**Pull request by alexeldeib**
this matches the recent RP // API change: https://github.com/Azure/azure-rest-api-specs/pull/12120

I will start using this field in future changes, like in https://github.com/Azure/AgentBaker/pull/414 Just wanted to break down the changes.

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -548,12 +548,18 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			if profile != nil && profile.KubernetesConfig != nil && profile.KubernetesConfig.ContainerRuntimeConfig != nil && profile.KubernetesConfig.ContainerRuntimeConfig[datamodel.ContainerDataDirKey] != "" {
 				return true
 			}
+			if profile.KubeletDiskType == datamodel.TempDisk {
```
Comment (alexeldeib) [inline pkg/agent/baker.go:551]: the thinking here is we'll remove ContainerRuntimeConfig. 

it ended up not being very useful, and actually we can't set arbitrary paths for this field anyway -- it needs to be predictable for monitoring and mounting symlinks properly into containers reading docker logs.

File: pkg/agent/datamodel/const.go [hunk 2]
```diff
@@ -98,3 +98,9 @@ const (
 	// AADPodIdentityAddonName is the name of the aad-pod-identity addon deployment
 	AADPodIdentityAddonName = "aad-pod-identity"
 )
+
+const (
+	// TempDiskContainerDataDir is the path used to mount docker images, emptyDir volumes, and kubelet data
+	// when KubeletDiskType == TempDisk.
+	TempDiskContainerDataDir = "/mnt/aks/containers"
```
Comment (alexeldeib) [inline pkg/agent/datamodel/const.go:105]: basically just need something under /mnt that is unique to aks. we will eventually put other things here, so I'm reserving the `/mnt/aks` prefix for all our content, and specifically taking `/mnt/aks/containers` for the container data dir.

File: pkg/agent/baker.go (hunk 1 above)
Comment (alexeldeib) [reply pkg/agent/baker.go:551]: there are no active users of the toggle anyway since we pushed ephemeral os :)

Comment (alexeldeib) [reply pkg/agent/baker.go:551]: (ContainerRuntimeConfig is how this value was propagated from the toggle)

File: pkg/agent/datamodel/const.go (hunk 2 above)
Comment (yizhang4321) [reply pkg/agent/datamodel/const.go:105]: If this is used for temp disk, would put temp disk somewhere in the name be helpful in the long run? /mnt/aks/containers sounds like something very official and important. If you want to reserve /mnt/aks for various use, how about /mnt/aks/kubelettempdisk?

**Review by yizhang4321 — APPROVED** [review]

File: pkg/agent/datamodel/const.go (hunk 2 above)
Comment (alexeldeib) [reply pkg/agent/datamodel/const.go:105]: /mnt is where Azure puts the temp disk by default, we are merely claiming a "slice" of /mnt for aks :)

Comment (alexeldeib) [reply pkg/agent/datamodel/const.go:105]: the subdirectories would be /mnt/aks/kubelet and /mnt/aks/containers for data

**Pull request merged**